src/storage/txn/commands/acquire_pessimistic_lock.rs

```diff
@@ -104,101 +105,96 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for AcquirePessimisticLock
                 self.min_commit_ts,
                 need_old_value,
                 self.lock_only_if_exists,
-                false,
+                self.allow_lock_with_conflict,
             ) {
                 Ok((key_res, old_value)) => {
-                    res.as_mut().unwrap().push(key_res);
+                    res.push(key_res);
                     if old_value.resolved() {
                         let key = k.append_ts(txn.start_ts);
                         // MutationType is unknown in AcquirePessimisticLock stage.
                         let mutation_type = None;
-                        self.old_values.insert(key, (old_value, mutation_type));
+                        old_values.insert(key, (old_value, mutation_type));
                     }
+                    written_rows += 1;
                 }
-                Err(e @ MvccError(box MvccErrorInner::KeyIsLocked { .. })) => {
-                    res = Err(e).map_err(Error::from).map_err(StorageError::from);
+                Err(MvccError(box MvccErrorInner::KeyIsLocked(lock_info))) => {
+                    let request_parameters = PessimisticLockParameters {
+                        pb_ctx: ctx.clone(),
+                        primary: self.primary.clone(),
+                        start_ts,
+                        lock_ttl: self.lock_ttl,
+                        for_update_ts: self.for_update_ts,
+                        wait_timeout: self.wait_timeout,
+                        return_values: self.return_values,
+                        min_commit_ts: self.min_commit_ts,
+                        check_existence: self.check_existence,
+                        is_first_lock: self.is_first_lock,
+                        lock_only_if_exists: self.lock_only_if_exists,
+                        allow_lock_with_conflict: false,
```

Comment:
Always Set `allow_lock_with_conflict` to false instead of self.allow_lock_with_conflict?

Reply:
Sorry, it's a mistake.